airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/http_extractor.py

```diff
@@ -3,13 +3,19 @@
 #
 
 from abc import ABC, abstractmethod
-from typing import List
+from typing import Any, List, Mapping
 
 import requests
 from airbyte_cdk.sources.declarative.types import Record
 
 
 class HttpExtractor(ABC):
     @abstractmethod
-    def extract_records(self, response: requests.Response) -> List[Record]:
+    def extract_records(
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
```

Comment:
@brianjlai open question: would it make more sense for the extractor to have access to the state object instead of passing in the stream_state, stream_slice, and next_page_token directly?
It seems like a more generic pattern we could apply to a few more classes